### PR TITLE
[Copy] Update job poster

### LINKED ISSUE #7245: ✨ Job poster changes
## ✨ Feature

A couple of minor copy changes to the job poster 

## 🕵️ Acceptance Criteria

changes to the following sections
- Location, language, and security
- Contact and accommodation
- Hiring policies 

Please note all changes have been marked with comments on the figma file 

## 📸 Screenshot [changes marked in red]
Screenshot 2023-07-12 at 3 00 45 PM
Screenshot 2023-07-13 at 9 20 13 AM

## 📍Location, language, and security

Change copy from "Remote optional" to "Remote, hybrid or on-site"

## 📧Contact and accommodation 

change copy from "Do you require accommodations? or do you have any questions about this process? Please contact the Digital Community Management Office if you require any accomodations during this application process.Email: fames@acanteipsum.ca" to be more general. 

New copy: "Please contact the [[Name] team](example@email.com) by email if you have **any questions** or **require an accommodation** during this process." 

## 💼 Hiring policies  

Change the title from "Hiring policies" to "Who can apply?" 
change body copy to "Persons residing in Canada, and Canadian citizens and Permanent residents abroad. 
Preference will be given to veterans, Canadian citizens, and to permanent residents." 

Please note that this title copy change should also be reflected in the nav bar

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7245 

## 👋 Introduction

Updates some copy on the job poster page.

## 🕵️ Details

This also hides the contact section if no email exists. This should never happen since email is required but was done to be safe since we are not meant to use any fallback mentioned in https://github.com/GCTC-NTGC/gc-digital-talent/issues/7219

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Ensure a post has "Remote" selected under the Location option
3. Visit the poster
4. Confirm copy changes match [the mockup ](https://www.figma.com/file/CdYmWlP6SpT17bw7Q9411Z/Candidates---job-poster?type=design&node-id=51-3179&mode=design&t=SScNmkymmIofUEBe-0)

## 📸 Screenshot

Screenshot 2023-07-14 082446

### CODE REVIEW

**Comment by vd1992** [comment]
Question

Looking at the screenshot in the issue body
The circled bit wasn't marked with a red square, but it isn't present on the poster either. Should it be?

image

**Comment by vd1992** [comment]
Got a translation bug, but that is independent of this issue I think, you didn't touch this part
Spinning off...

image

**Comment by esizer** [comment]
> Looking at the screenshot in the issue body
> The circled (oval) bit wasn't marked with a red square, but it isn't present on the poster either. Should it be?

Oh yeah, we actually don't have a URL for that yet so that is why.

File: apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx [hunk 1]
```diff
@@ -665,55 +668,46 @@ export const PoolPoster = ({
...
               </TableOfContents.Heading>
               <Text>
                 {intl.formatMessage({
                   defaultMessage:
-                    "Preference will be given to veterans, Canadian citizens and to permanent residents.",
-                  id: "IF1xj8",
+                    "Persons residing in Canada, and Canadian citizens and Permanent residents abroad.",
+                  id: "DomrM8",
+                  description: "List of criteria needed in order to apply",
+                })}
```
Comment (vd1992) [inline apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx:704]: Should the casing vary like this? This is the same as in the issue but is it a typo?

image

Comment (esizer) [reply apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx:704]: Hmm, yeah that makes sense. @mnigh Any idea how we would prefer this to be cased?

Comment (mnigh) [reply apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx:704]: permanent resident should be sentence case, so all lower case for these examples.

Comment (esizer) [reply apps/web/src/pages/Pools/PoolAdvertisementPage/PoolAdvertisementPage.tsx]: Thanks, updated in [fix case inconsistency](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7287/commits/8e20bbc97c4f4ecbadf3a015d1fb97a21fd7938c)

**Pull request merged**